senator/game.py

```diff
@@ -78,45 +97,93 @@ def nash_conv(self) -> float:
         nash_convexity = 0
         for player in range(self._num_players):
             instantaneous_regret = self._get_instaneous_regret(player, strategy)
-            nash_convexity += np.sum(instantaneous_regret * strategy[player])
+            assert np.max(instantaneous_regret) >= 0
+            nash_convexity += np.max(instantaneous_regret)
 
         return nash_convexity
 
 
 class Game:
-    def __init__(self, initial_utility: np.ndarray, votes: np.ndarray):
-        assert len(initial_utility.shape) == 2, "Initial utility should be a 2D array"
+    def __init__(
+        self,
+        initial_utility: np.ndarray,
+        votes: np.ndarray,
+        is_owner: np.ndarray,
+        owner_loss: float = 1,
+        other_loss: float = 0,
+        owner_trashold: float = 0,
+    ) -> None:
+
+        self.owner_loss = owner_loss
+        self.other_loss = other_loss
+        self.owner_trashold = owner_trashold
+        self.is_owner = is_owner
         self.votes = votes
         self.initial_utility = initial_utility
-        root_utility_matrix = get_utility_matrix(initial_utility, votes)  # TODO: see what it takes to get the utility matrix
+        root_utility_matrix = get_utility_matrix(initial_utility, votes, is_owner)
         self.root = Node(parent=None, utility_matrix=root_utility_matrix)
 
-    def add_child(self, node: Node, joint_action: np.ndarray, current_utility: np.ndarray, votes: np.ndarray) -> None:
-        child_utility_matrix = get_utility_matrix(current_utility, votes)  # TODO: see what it takes to get the utility matrix
-        node.children[tuple(joint_action)] = Node(parent=node, utility_matrix=child_utility_matrix)
-
-    def solve_node_via_rm_plus(self, node: Node, iterations: int = 1000) -> None:
+    def add_child(
+        self,
+        node: Node,
+        joint_action: np.ndarray,
+        current_utility: np.ndarray,
+        votes: np.ndarray,
+    ) -> None:
+        child_utility_matrix = get_utility_matrix(
+            current_utility,
+            votes,
+            self.is_owner,
+            self.owner_loss,
+            self.other_loss,
+            self.owner_trashold,
+        )  # TODO: see what it takes to get the utility matrix
+        node.children[tuple(joint_action)] = Node(
+            parent=node, utility_matrix=child_utility_matrix
+        )
+
+    def solve_node_via_rm_plus(self, node: Node, iterations: int = 2000) -> None:
```

Comment:
Těch 2000 iterací je možná trochu random. Třeba některé utility matice budou potřebovat mín a některé víc. Možná by se tady dalo udělat něco jako 
`while node.nash_conv > 0 and iterations < 1e4:`
a tam házer rm_plus_step?

Reply:
Já jsem večer pustil experimenty na 10 000 iterací, protože pořád něco padalo a stejně se našla nastavení parametrů, kde Nash Conv padl... 

Takhle to nakombinovat dobrý nápad.

Reply:
Ikdyž teda našel jsem i kombinaci parametrů, kde to neseběhlo za 10 000 iterací ani pro rm_plus_step, ale je jich fakt něco jako 1-3%.